plugins/ptp_operator/metrics/logparser.go

```diff
@@ -297,7 +297,7 @@ func extractPTP4lEventState(output string, isFollowerOnly bool) (portID int, rol
 		strings.Contains(output, "SLAVE to UNCALIBRATED") ||
 		strings.Contains(output, "MASTER to UNCALIBRATED on RS_SLAVE") ||
 		strings.Contains(output, "LISTENING to UNCALIBRATED on RS_SLAVE") ||
-		(strings.Contains(output, "FAULTY to LISTENING on INIT_COMPLETE") && !isFollowerOnly) { // added to manage two port case so its not breaking
+		strings.Contains(output, "FAULTY to LISTENING on INIT_COMPLETE") { // added to manage two port case so its not breaking
```

Comment:
Shouldn't this transition lead to LISTENING? Why is it FAULTY? the same transition is on line 310. Imho, we should process the LISTENING condition to also trigger the HOLDOVER if needed

Reply:
the role is exposed via metrics, so in this case a LISTENING role would be FAULTY. Would be simpler to not change the roles. The LISTENING is a fault only if there is not other port that is a SLAVE, so in the general case LISTENING is not necessarily FAULTY

Reply:
We can work on this refactor in a new PR. I will accept these changes.